File: langs/julia/prime.jl
function sieve(size)
    sieve_array = falses(size)
    if size <= 0
        return sieve_array
    end
    sieve_array[1] = true

    root = floor(Int, sqrt(size) + 1)
    @inbounds for i = 2 : root
        if !sieve_array[i]
            sieve_array[i * i : i : size] = true
        end
    end
    return sieve_array
end

function count_prime(n)
    return count(x -> !x, sieve(n))
end

n = 10000000
if length(ARGS) >= 1
    try
        n = parse(Int, ARGS[1])
    end
end
println(count_prime(n))

File: langs/rust/prime.rs
#![feature(step_by)]
use std::cmp;
use std::env;

fn sieve(size: usize) -> Vec<bool> {
    let mut sieve_array = vec![true; size];
    for i in 0..cmp::min(2, size) {
        sieve_array[i] = false;
    }

    let root: usize = ((size as f64).sqrt() as usize) + 1;
    for i in 2..root {
        if sieve_array[i] {
            for j in ((i * i)..size).step_by(i) {
                sieve_array[j] = false;
            }
        }
    }
    
    return sieve_array;
}

fn count_prime(n: usize) -> usize {
    return sieve(n + 1).iter().filter(|&x| *x).count();
}

fn main() {
    let n = env::args().nth(1)
      .and_then(|s| usize::from_str_radix(s.as_str(), 10).ok())
      .unwrap_or(10000000usize);
    println!("{}", count_prime(n));
}
